src/Control/Session.php

```diff
@@ -354,6 +368,17 @@ public function start(HTTPRequest $request)
         $this->started = true;
     }
 
+    /**
+     * Determines what the value for the `secure` cookie attribute should be.
+     */
+    private function isCookieSecure(string $sameSite, bool $isHttps): bool
+    {
+        if ($sameSite === 'None') {
+            return true;
```

Comment:
What happens when $isHttps is false?

Reply:
According to the spec, `secure` _must_ be true when `samesite` is "None". It doesn't specify how or whether this interacts in some way with the `httponly` attribute.
In short: I don't know, but I don't think it's relevant to this return value.

Reply:
That makes sense, you'll need to update docs + unit tests accordingly

Reply:
This change requires keeping the private `getSameSite()` method in the next major, so it can't be deprecated anymore. I've created #10342 to capture the intention for adding a `$sameSite` parameter in the next major.

Reply:
We've talked a bit offline about how to handle this situation. There's a chance that `Director::is_https()` incorrectly returns false where https is terminated at a load balancer or other proxy. Because of this, we don't want to throw exceptions - that would be causing errors for actually valid situations. Instead, we want to log a warning (in case it is actually incorrect configuration) but let the cookie be set.
The browser then will reject the cookie (with a warning in the console) if it is indeed invalid, but will accept the cookie if it is valid.